File: Competition/2019/Golden/Weighted-Sum/solution.py


def main():
    
    digits, weights = get_n_process_STDIN_data()
    weighted_sum = 0

    for i in range(len(digits)):

        product = digits[i] * weights[i]
        weighted_sum += product

    print(f"weighted sum is {weighted_sum}")

def get_n_process_STDIN_data():

    print("Enter the number:")
    num = input()
    weight = 0

    digits = []
    weights = []

    for char in num:
        weight += 1
        digits.append(int(char))
        weights.append(weight)
    
    return digits, weights

main()

File: Competition/2019/Golden/Weighted-Sum/Solution.java
import java.util.ArrayList;
import java.util.Scanner;

public class Solution
{
    public static void main(String[] args)
    {
        Scanner myScanner = new Scanner(System.in);

        System.out.println("Enter the number");
        String numStrForm = myScanner.nextLine();
        myScanner.close();

        int weight = 0;
        int weightedSum = 0;

        ArrayList<Integer> digits = new ArrayList<Integer>();
        ArrayList<Integer> weights = new ArrayList<Integer>();

        for (char lilChar : numStrForm.toCharArray())
        {
            weight++;
            int digit = Integer.valueOf(String.valueOf(lilChar));
            digits.add(digit);
            weights.add(weight);

        }

        for (int i = 0; i < digits.size(); i++)
        {
            int product = digits.get(i) * weights.get(i);
            weightedSum += product;
        }

        System.out.println("weighted sum is " + weightedSum);

    }
}